packages/data/src/plugins/async-generator/index.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { applyMiddleware } from 'redux';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import asyncGeneratorMiddleware, { toAsyncIterable } from '../../middlewares/async-generator';
+
+export default function( registry ) {
+	return {
+		registerStore( reducerKey, options ) {
+			const store = registry.registerStore( reducerKey, options );
+			const enhancer = applyMiddleware( asyncGeneratorMiddleware );
+			const createStore = () => store;
+			Object.assign(
+				store,
+				enhancer( createStore )( options.reducer )
+			);
+			return store;
+		},
+
+		async fulfill( reducerKey, selectorName, ...args ) {
+			const resolver = get( registry.namespaces, [ reducerKey, 'resolvers', selectorName ] );
+			if ( ! resolver ) {
+				return;
+			}
+			const store = registry.namespaces[ reducerKey ].store;
+			const state = store.getState();
+			const actionCreator = resolver.fulfill( state, ...args );
```

Comment:
Would `fulfill` (the function, not the result) be considered the action creator? And what we're assigning to be the `action` ?

Reply:
The action creator is `resolver.fulfill` yes and I'm not sure I understand the second part of the question.

In this particular case though (async generator), we can't consider a resolver as an action creator because the resolvers takes the state as first argument (and action creators don't) and that's exactly why I'm overriding the default `fulfill` function of the registry.